crates/ra_syntax/src/grammar.ron

```diff
@@ -153,6 +154,7 @@ Grammar(
         "FOR_TYPE",
         "IMPL_TRAIT_TYPE",
         "DYN_TRAIT_TYPE",
+        "EXISTENTIAL_TYPE",
```

Comment:
Ah, one more thing!

I don't think we need to create a separate `EXISTENTIAL_TYPE` node type. Rather, we should re-use existing `TYPE_ALIAS`, and treat existential as a modifer. So, `type_alias_def` should still call `m.complete(p, TYPE_ALIAS_DEF);` inside

Reply:
when refers to something `alias`, `=` is a must. I am a bit concern to treat `EXISTENTIAL_TYPE` as a `special` case of `alias`.

Reply:
I think that's OK: we already don't require `=` in types in traits:

```
trait Future {
    type Output;
}
```

Reply:
okay, I'm persuaded :)